common/src/main/java/io/netty/util/internal/CleanerJava6.java

```diff
@@ -65,41 +80,24 @@ public Object run() {
                 throw (Throwable) mayBeCleanerField;
             }
 
-            cleanerField = (Field) mayBeCleanerField;
-
-            final Object cleaner;
-
-            // If we have sun.misc.Unsafe we will use it as its faster then using reflection,
-            // otherwise let us try reflection as last resort.
-            if (PlatformDependent.hasUnsafe()) {
-                fieldOffset = PlatformDependent0.objectFieldOffset(cleanerField);
-                cleaner = PlatformDependent0.getObject(direct, fieldOffset);
-            } else {
-                fieldOffset = -1;
-                cleaner = cleanerField.get(direct);
-            }
-            clean = cleaner.getClass().getDeclaredMethod("clean");
-            clean.invoke(cleaner);
+            clean = (MethodHandle) mayBeCleanerField;
+            clean.invokeExact(direct);
```

Comment:
I've never played with method handle like this (always used `invokeExact` on unreflected stuff) - let's check the benchmarks to see if it is as  expected!

Reply:
Profiling shows the method handles add less than 1% overhead on top of the `Cleaner.clean()` call, so I think we're good.